main.go

```diff
@@ -2,31 +2,90 @@ package main
 
 import (
 	"context"
+	"encoding/base64"
+	"net/http"
 	"os"
+	"strconv"
+	"time"
 
+	"github.com/asecurityteam/nexpose-vuln-notifier/pkg/assetfetcher"
+	nexposevulnnotiifier "github.com/asecurityteam/nexpose-vuln-notifier/pkg/handlers/v1"
+	"github.com/asecurityteam/runhttp"
 	serverfull "github.com/asecurityteam/serverfull/pkg"
 	serverfulldomain "github.com/asecurityteam/serverfull/pkg/domain"
 	"github.com/asecurityteam/settings"
+	"github.com/asecurityteam/transport"
 	"github.com/aws/aws-lambda-go/lambda"
 )
 
 func main() {
 	ctx := context.Background()
-	var _ lambda.Handler = nil // Placeholder to keep lambda imported. Delete after adding to the map.
-	handlers := map[string]serverfulldomain.Handler{
-		// TODO: Register lambda functions here in the form of
-		// "name_or_arn": lambda.NewHandler(myHandler.Handle)
+
+	nexposeHTTPClient := nexposeHTTPClient()
+
+	pageSize, _ := strconv.Atoi(os.Getenv("NEXPOSE_SITE_ASSET_PAGE_SIZE"))
+
+	notifier := &nexposevulnnotiifier.NexposeVulnNotificationHandler{
+		AssetFetcher: &assetfetcher.NexposeAssetFetcher{
+			HTTPClient: nexposeHTTPClient,
+			Host:       os.Getenv("NEXPOSE_HOST"),
+			PageSize:   pageSize,
+		},
+		LogFn:  runhttp.LoggerFromContext,
+		StatFn: runhttp.StatFromContext,
+	}
+
+	lambdaHandlers := map[string]serverfulldomain.Handler{
+		"notification": lambda.NewHandler(notifier.Handle),
 	}
 
 	source, err := settings.NewEnvSource(os.Environ())
 	if err != nil {
 		panic(err.Error())
 	}
-	rt, err := serverfull.NewStatic(ctx, source, handlers)
+
+	rt, err := serverfull.NewStatic(ctx, source, lambdaHandlers)
 	if err != nil {
 		panic(err.Error())
 	}
 	if err := rt.Run(); err != nil {
 		panic(err.Error())
 	}
 }
+
+func nexposeHTTPClient() *http.Client {
+	nexposeTimeout, err := strconv.Atoi(os.Getenv("NEXPOSE_REQUEST_TIMEOUT_MS"))
+	if err != nil {
+		panic(err.Error())
+	}
+	var retryDecorator = transport.NewRetrier(
+		transport.NewPercentJitteredBackoffPolicy(transport.NewFixedBackoffPolicy(50*time.Millisecond), .25),
+		transport.NewLimitedRetryPolicy(3,
+			transport.NewStatusCodeRetryPolicy(http.StatusInternalServerError, http.StatusBadGateway, http.StatusGatewayTimeout),
+			transport.NewTimeoutRetryPolicy(time.Duration(nexposeTimeout)*time.Millisecond)),
+	)
+
+	var headerDecorator = transport.NewHeader(
+		func(*http.Request) (string, string) {
+			return "Authorization", basicAuth(os.Getenv("NEXPOSE_USERNAME"), os.Getenv("NEXPOSE_PASSWORD"))
```

Comment:
I wonder if it would be worth adding a basic auth component to transportd since it is so common.

Reply:
I think so. I can create an issue for it